src/bridge/IrcBridge.ts

```diff
@@ -183,8 +185,22 @@ export class IrcBridge {
 
     private initialiseMetrics() {
         const zeroAge = new PrometheusMetrics.AgeCounters();
+        const registry = new Registry();
 
-        const metrics = this.bridge.getPrometheusMetrics();
+        const usingRemoteMetrics = !!this.config.ircService.metrics.port;
+
+        const metrics = this.bridge.getPrometheusMetrics(!usingRemoteMetrics, registry);
+
+        if (this.config.ircService.metrics.port) {
+            log.info(`Spawned metrics worker on http://${this.config.ircService.metrics.host}:${this.config.ircService.metrics.port}`);
+            spawnMetricsWorker(
+                this.config.ircService.metrics.port,
+                this.config.ircService.metrics.host,
+                () => {
+                    return registry.metrics();
```

Comment:
Can the third parameter just be `registry.metrics`?
I don't see why it needs to be wrapped in a function. Please use `.bind()` if that's the intention behind this.